docs/cudf/source/developer_guide/ARCHITECTURE.md

```diff
@@ -0,0 +1,208 @@
+# cuDF Architecture
+
```

Comment:
I think it's worth mentioning very early on that cuDF wraps a  C++ library called `libcudf`, which does most of the actual computation. Also perhaps it's a good idea to point to the libcudf design doc here for folks curious to learn how _it_ works.

We introduce the term `libcudf` below without really saying what exactly it is.